### PR TITLE
Add update orphaning dashboard.

### PR DESCRIPTION
**Pull request by sapohl**
This is to add a preliminary update orphaning dashboard to telemetry.mozilla.org, which is hard-coded to one weekly report. This dashboard relies on longitudinal datasets. Once longitudinal datasets are created automatically, a scheduled Spark job will create weekly .json files for this dashboard. Once we have a weekly Spark job, I will create a follow-up pull-request to change the dashboard to handle these weekly .json files, instead of being hard-coded to one weekly report.

### CODE REVIEW

File: update-orphaning/index.html [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+  margin: 1px;
+  color: white;
+}
+
+#update-ping-code-description {
+  height: 50px;
+  border-radius: 5px;
+  border: 2px solid grey;
+  padding: 20px; 
+  /*width: 200px;*/
```
Comment (chutten) [inline update-orphaning/index.html]: omit

File: update-orphaning/index.html [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+  if (date < 10) {
+    date = "0" + date;
+  }
+  var friAfterLastWedDate = new Date();
+  friAfterLastWedDate.setDate(lastWedDate.getDate() + 2);
+  var lastWedDataFile = year + month + date + ".json";
+  var startDate = new Date();
+  //startDate.setDate(1);
+  //startDate.setMonth(0);
+  //startDate.setFullYear(2015);
```
Comment (chutten) [inline update-orphaning/index.html]: More commented-out code that can be omitted

File: update-orphaning/index.html [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+      tempDate = "0" + tempDate;
+    }
+    dateDropdown.innerHTML += "<option value=\"" + tempYear + tempMonth + tempDate + ".json\">" + tempMonth + "/" + tempDate + "/" + tempYear + "</option>";
+    tempWedDate.setDate(tempWedDate.getDate() - 7);
+  }
+
+  var currentlySelectedDataFile = lastWedDataFile;
+
+  //var testDiv = document.getElementById("test");
+  //testDiv.innerHTML = lastWedDataFile;
```
Comment (chutten) [inline update-orphaning/index.html]: omit

File: update-orphaning/index.html [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+    {
+      document.getElementById("out-of-date-of-concern-summary").innerHTML = "<strong>No data found!</strong><br>(Failing URL: " + dataUrl + branch + dataFile + ")";
+      document.getElementById("out-of-date-updates-disabled-summary").innerHTML = "";
+      document.getElementById("up-to-date-summary").innerHTML = "";
+      document.getElementById("distribution-chart").innerHTML = "";
+      document.getElementById("update-ping-codes-chart").innerHTML = "";
+      document.getElementById("update-ping-code-description").innerHTML = "Hover over a bar below to see a description of the ping code.";
+    }
+
+    $.getJSON("https://analysis-output.telemetry.mozilla.org/spohl/update-orphaning-data/" + dataFile,{}, function(data) {
```
Comment (chutten) [inline update-orphaning/index.html]: nit: there should be a space before `{}`

File: update-orphaning/index.html [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+        var totalValues = 0;
+        for (i = -1; i <= 34; i++) {
+          var val = 0;
+          if (updateChecks[i]) {
+            val = updateChecks[i];
+          }
+          updateCodesData.push({index:i,value:val});
+          totalValues += val;
+        }
+        //console.log(updateCodesData);
```
Comment (chutten) [inline update-orphaning/index.html]: omit

File: update-orphaning/index.html [hunk 6]
```diff
@@ -0,0 +1,462 @@
...
+        var margin = {top: 20, right: 20, bottom: 30, left: 40},
+                      width = 960 - margin.left - margin.right,
```
Comment (chutten) [inline update-orphaning/index.html]: It is good practice to repeat `var` on each line as before

**Comment by chutten** [comment]
Was the version of d3pie in [new-pipeline/src](https://github.com/mozilla/telemetry-dashboard/tree/gh-pages/new-pipeline/lib) incorrect for your use?

Also: is it deliberate that you are obfuscating this dashboard by not linking to it from the index?

**Comment by sapohl** [comment]
@vitillo - just wanted to make you aware of this pull request.

**Comment by sapohl** [comment]
@chutten, I believe all your comments have been addressed. Is there anything else that I need to do, or is there someone else who should review this? What's the next step to get this merged? Thank you.

**Comment by chutten** [comment]
Ack, does Github not send notifications when commits are added to pull requests, or just when comments are added? 

You did indeed address my concerns, and I've poked around at the live version [here](http://sapohl.github.io/telemetry-dashboard/update-orphaning/) which seems to hold up to casual scrutiny. I'm happy to land this.

**Comment by sapohl** [comment]
Thank you!

**Pull request merged**